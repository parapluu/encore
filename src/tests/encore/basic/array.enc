-- Repeated index operations should parse!
def jaggedIndexing(): void
  let a = new Array<Array<int>>(1) in
    {
      a[0] = new Array<int>(1);
      a[0][0] = 3;
      print a[0][0]
    }


def multiDimAcces(): void
  let n = 2
      m = 3
      arr = new Array2<int>(n, m)
  in {
    repeat i <- |arr|
      repeat j <- |arr[i]|
        arr[i,j] = m * i + j;
    repeat i <- |arr|
      repeat j <- |arr[i]|
	print arr[i,j]
  }

def multiDimLiteral(): void
  let a = [["one", "two"], ["three", "four"], ["five", "six"]] in
    repeat i <- |a|
      repeat j <- |a[i]|
        print a[i,j]
	
class Foo
  def return() : [int]
    new [int](10)
  def foo(m:Main) : void
    let
      a = this.return()
    in {
      repeat i <- 10
        a[i] = i;
      m ! continue(a)
    }

class Main
  def continue(a:[int]) : void
    let
        b = ["foo", "bar"]
    in {
      repeat i <- |a|
        print a[i];
      print b[0];
      print b[1];
    }

  def main() : void
    let
      x = new Foo
    in {
      jaggedIndexing();
      multiDimAcces();
      multiDimLiteral();
      x ! foo(this)
      }
