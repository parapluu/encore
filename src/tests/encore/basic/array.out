3
0
1
2
3
4
5
one
two
three
four
five
six
0
1
2
3
4
5
6
7
8
9
foo
bar
